duckduckgo/macos-browser
# Address bar redesign

Task/Issue URL: https://app.asana.com/0/1199237043630360/1199211719949610
Tech Design URL: https://www.figma.com/file/CNZfYqhO9GsZ4UqCleGT9w/Address-Bar?node-id=2%3A2350
CC: @brindy

**Description**:
This PR contains a Address Bar implementation accorting to the Design and bug fixes discussed in https://app.asana.com/0/1199237043630360/1199878361042097

**Steps to test this PR**:
1. Activate/deactivate Address Bar
2. Input some query, wait for suggestions to appear
3. Mouse-over suggestions, use up/down keys to choose suggestion
4. Click suggestion or use Enter to choose suggestion
5. Activate Address Bar when website is open (note icon)
6. Activate Address Bar when search is open (note icon)
7. Use Clear button
8. Clear address bar by repeatedly pressing Backspace
9. Use Refresh Button
10. Enter/exit fullscreen
11. Deactivate App when Address Bar is active; When suggestions are show
12. Activate app by clicking inactive Address Bar; By clicking active Address Bar
13. ...

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
